### ISSUE TITLE
Use drupol/phposinfo

### ISSUE THREAD

**Issue by drupol** [vaimo/composer-patches#36]
It seems that tivie/php-os-detector is a dependency and it can produces false results, see the corresponding issue: https://github.com/tivie/php-os-detector/issues/3

Based on this, I've created my own package: [drupol/phposinfo](https://packagist.org/packages/drupol/phposinfo) that fixes the issue.

This issue is to propose the replacement of tivie/php-os-detector with drupol/phposinfo.

**Comment by allanpaiste** [comment]
I'm afraid that we still can't use it as a replacement. 

As was stated once before - the package is kept on very low PHP version dependency on purpose due to the most uses of said plugin coming from being used against older software that does not have active maintainer any more (otherwise one would expect issues or feature requests to be addressed directly in the distributed releases).

Furthermore, I ran phpcs with compatibility ruleset against 1.6.1.1, which states to be for PHP 5.6, this is what I got:

```
FILE: /home/vagrant/phposinfo/src/OsInfo.php
----------------------------------------------------------------------
FOUND 3 ERRORS AFFECTING 3 LINES
----------------------------------------------------------------------
 179 | ERROR | The constant "PHP_OS_FAMILY" is not present in PHP
     |       | version 7.1 or earlier
 180 | ERROR | The constant "PHP_OS_FAMILY" is not present in PHP
     |       | version 7.1 or earlier
----------------------------------------------------------------------

FILE: /home/vagrant/phposinfo/src/Enum/Enum.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 24 | ERROR | "yield from" keyword (for generators) is not present in
    |       | PHP version 5.6 or earlier
----------------------------------------------------------------------
```

Ideally, I'd of course encourage making it compatible with 5.3, which is not that far off:

```
FILE: /home/vagrant/phposinfo/src/OsInfo.php
----------------------------------------------------------------------
FOUND 5 ERRORS AFFECTING 5 LINES
----------------------------------------------------------------------
 197 | ERROR | Short array syntax (open) is available since 5.4
 200 | ERROR | Short array syntax (close) is available since 5.4
----------------------------------------------------------------------


FILE: /home/vagrant/phposinfo/src/Enum/Enum.php
----------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
----------------------------------------------------------------------
 18 | ERROR | The magic class constant ClassName::class was not
    |       | available in PHP 5.4 or earlier
----------------------------------------------------------------------
```

I understand that you want the utility to be a good replacement for the current one that can produce faulty results, but the idea of having patches against packages has to remain backwards compatible with older systems/products.

**Comment by drupol** [comment]
Hello,

I pushed a commit that reduce the minimum version of PHP to ^5.3.

A tag ha been created: 1.6.1.2

Let me know if it's better for you now :-)

**Comment by allanpaiste** [comment]
Looks cool! :) Thank you!

Change done and released in 3.49.0 & 4.18.0

**Comment by drupol** [comment]
Thanks !!!
